task10/index.js

```diff
@@ -0,0 +1,63 @@
+const div = document.getElementById("js-parent");
+
+const loading = () => {
+    const loadingPlace = document.createElement("div");
+    const gif = document.createElement('img');
+    loadingPlace.id = "loadingPlace";
+    gif.src = "loading-circle.gif";
+    div.appendChild(loadingPlace).appendChild(gif);
+};
+
+const removeLoading = () => {
+    document.getElementById("loadingPlace").remove();
+};
+
+const getData = () => {
+    const obj = [
+        {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+        {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+    ]
+    
+    return new Promise((resolve,reject) => {
+        setTimeout(() => {
+            resolve(obj);
+            reject("error!");
+        },2000);
+    });    
+};
+
+const objShow = async () => {
+    loading();
+    
+    const value = await getData();
+    const ul = document.createElement('ul');
+    const fragment = document.createDocumentFragment();
+
+    value.forEach(item => {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.textContent = item.text;
+        a.href = item.to;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    })
+    div.appendChild(ul).appendChild(fragment);
+    
+};
+objShow();
+
+const exception = async () => {
+    try {
+        await getData();
+    } catch (error) {
+        console.error(error);
+    }
```

Comment:
This is just an IMO, it's up to you to fix it or not.
（When there is an error, the loading image is lost and nothing is displayed on the screen, so...）
IMO
```suggestion
    } catch (error) {
        div.textContent = "データが取得できませんでした";
        console.error(error);
    }
```

Reply:
This writing style is more friendly. Fixed.